KevinHardy-Cooper/VApp
# Add Testing Framework

# Description
This is the initial pull request to establish a testing framework for our code. We will be using MochaJS, a mature and supported framework.

In this PR, all I have done is added a few functional test cases for the User module. The first test case is provided by MochaJS as a sample, so I left it in for reference.

Note that the way the test is set up, there will be a new user called "test@email.com" created every time you run the tests. I think usually this would be okay but since we don't have a deleteUser() function or anything, we just have to be mindful not to run tests on production. Not that it really matters though...

Ideally, as new features/functions/etc. are added and completed, the associated test cases will also be done. We will need to spend some time to implement all (or most) of the test cases identified in the V&V before the final presentation.

# How to Test
- Pull and checkout the branch locally
- Run `npm install` to install MochaJS, the testing framework
- Run `npm test` to run the test cases identified in the `test/test.js` file
- The test cases should run successfully

# New Dependencies
- [MochaJS](https://mochajs.org/)
